### ISSUE TITLE
Syncing to an encrypted Seafile library stops after 1 hour

### ISSUE THREAD

**Issue by Fogelholk** [rclone/rclone#6662]
#### What is the problem you are having with rclone?
When syncing files with the Seafile backend to an encrypted library the sync just stops with error messages after 1 hour. Encrypted libraries in Seafile are only decrypted in a web browser for 1 hour until you need to input the encrypted password again, seems like this unfortunately also affects rclone.

https://help.seafile.com/security_and_encryption/use_encrypted_libraries/
- If you use web app, you have to input the password to the server. The server will cache the password in encrypted format for 1 hour. It won't store the password on disk.

I haven't found out a way to increase this 1 hour limit on the Seafile server-side, except not using an encrypted library. However I want to use an encrypted library to transfer my files into.

Anyhow, after exactly an hours of successfully syncing files without any issues (except a few one-byte-off transfers, which are probably fixed by re-sync if the sync didn't stop after 1 hour), the following errors show up after about 1 minute timeout for every file still not transferred:
```
Dec 30 10:48:52 rclone rclone[56954]: file1: Failed to copy: internal server error during file upload
Dec 30 10:49:02 rclone rclone[56954]: file2: Failed to copy: internal server error during file upload
```

#### What is your rclone version (output from `rclone version`)
```
# rclone --version
rclone v1.59.2-DEV
- os/version: freebsd 13.1-release-p2 (64 bit)
- os/kernel: 13.1-release-p2 (amd64)
- os/type: freebsd
- os/arch: amd64
- go/version: go1.19.2
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
```
FreeBSD 13.1-RELEASE-p5, in an iocage jail on TrueNAS-13.0-U3
```

#### Which cloud storage system are you using? (e.g. Google Drive)
```
Seafile backend
```

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```
rclone --config=/usr/local/etc/rclone.conf --transfers 8 --stats-log-level NOTICE --syslog --ignore-case --exclude "thumbnails/**" --delete-excluded sync /mnt/SOURCE/ rclone-seafile:
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
Dec 30 10:48:52 rclone rclone[56954]: file1: Failed to copy: internal server error during file upload
Dec 30 10:49:02 rclone rclone[56954]: file2: Failed to copy: internal server error during file upload
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by animosity22** [comment]
Can you update to the latest version and re-test please?

**Comment by Fogelholk** [comment]
Didn't see any updates to Seafile between the versions, and I'm following the quarterly branch on FreeBSD hence the ~3 months old version. Anyways, built rclone from ports but still the same problems:

```
# rclone --version
rclone v1.61.1-DEV
- os/version: freebsd 13.1-release-p2 (64 bit)
- os/kernel: 13.1-release-p2 (amd64)
- os/type: freebsd
- os/arch: amd64
- go/version: go1.19.4
- go/linking: dynamic
- go/tags: none
```

After exactly 1 hour files just stop transferring and I get the same error:

```
Jan  1 18:13:20 rclone rclone[22179]: xxx: Failed to copy: internal server error during file upload
Jan  1 18:13:23 rclone rclone[22179]: xxx: Failed to copy: internal server error during file upload
Jan  1 18:13:28 rclone rclone[22179]: xxx: Failed to copy: internal server error during file upload
```

**Comment by ncw** [comment]
> internal server error during file upload

This error message is from Seafile rather than rclone, so is possibly a seafile bug?

@creativeprojects do you have any ideas about this?

**Comment by creativeprojects** [comment]
We probably need to re-post the library key after an hour to reload the decryption key in the server. Either automatically some time before an hour or after the first error, assuming the server returns a specific error code (decryption error).

I'll have a look how it handles a re-post of the library key while a file is being transferred, otherwise we would need to pause any transfer momentarily 👎🏻

**Comment by creativeprojects** [comment]
That's what I thought, the server returns a generic error 😞 

`DEBUG : pacer: low level retry 1/1 (error HTTP error 500 (500 Internal Server Error) returned body: "{\"error\": \"Internal error\n\"}")`

so it will need to be sent in the background after some time

**Comment by ncw** [comment]
:-) You want something like this https://github.com/rclone/rclone/blob/master/lib/oauthutil/renew.go which is the same solution for the oauth backends but with your own logic.

**Comment by creativeprojects** [comment]
Cool, I had a similar draft implementation but I was wondering how to stop the timer in a nice way at the end. I can see how the Start/Stop does the trick 👍🏻 
Nice one

**Comment by ncw** [comment]
The fix for this is merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.62

**Comment by ncw** [comment]
Thank you @creativeprojects for fixing it :-)
